pkg/nar/util.go

```diff
@@ -7,3 +7,26 @@ import "strings"
 func IsValidNodeName(nodeName string) bool {
 	return !strings.Contains(nodeName, "/") && !strings.ContainsAny(nodeName, "\u0000")
 }
+
+// isLexicographicallyOrdered checks if two paths are lexicographically ordered component by component.
+func isLexicographicallyOrdered(path1 string, path2 string) bool {
```

Comment:
Maybe you can also call this PathisLexicographicallyOrdered, and just write a unit test for this, covering the various branches.

Reply:
Done!